src/signals/incident-management/containers/IncidentDetail/components/DetailHeader/components/DownloadButton/index.test.js

```diff
@@ -1,44 +1,69 @@
 import React from 'react';
-import { shallow } from 'enzyme';
+import { render, fireEvent } from '@testing-library/react';
 
 import DownloadButton from './index';
 
 describe('<DownloadButton />', () => {
   let props;
 
   beforeEach(() => {
-    const mockFetchPromise = new Promise((resolve) => resolve({
-      blob: () => new Blob()
-    }));
-
     props = {
       label: 'PDF',
-      url: 'zzz',
-      filename: '12345.pdf'
+      url: 'https://api.data.amsterdam.nl/signals/v1/private/signals/3077/pdf',
+      filename: 'SIA melding 3077.pdf'
     };
-    global.fetch = jest.fn().mockImplementation(() => mockFetchPromise);
-    global.window.URL.createObjectURL = jest.fn().mockImplementation(() => 'blob-href');
-  });
 
-  afterEach(() => {
-    global.fetch.mockClear();
-    delete global.fetch;
+    global.window.fetch = jest.fn();
+    global.window.URL.createObjectURL = jest.fn();
+    global.window.URL.revokeObjectURL = jest.fn();
   });
 
   describe('rendering', () => {
     it('should render correctly', () => {
-      const wrapper = shallow(
+      const { queryByTestId } = render(
         <DownloadButton {...props} />
       );
-      expect(wrapper).toMatchSnapshot();
+
+      expect(queryByTestId('download-button')).toHaveTextContent(/^PDF$/);
     });
+  });
+
+  describe('events', () => {
+    beforeEach(() => {
+      const res = new Response('{"hello":"world"}', {
+        status: 200,
+        headers: {
+          'Content-type': 'application/blob',
+        },
+      });
 
-    it('should render correctly when logged in', () => {
-      props.accessToken = 'access-token';
-      const wrapper = shallow(
+      global.window.fetch.mockReturnValue(Promise.resolve(res));
+    });
+
+    it('should download document', () => {
```

Comment:
The test name is confusing; the test case doesn't actually verify that a document is downloaded. Only that `fetch` is called.